File: 141-LinkedListCycle/141-LinkedListCycle.py
# Last updated: 2/28/2026, 4:28:26 PM
class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
    
        fast = head
        slow = head
        
        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next
            
            if fast == slow:
                return True
    
        return False

File: 141-LinkedListCycle/141-LinkedListCycle.java
// Last updated: 2/28/2026, 4:28:31 PM
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode p,q;

        if(head == null) return false;

        p = head;
        q = head.next;

        if(p == null || q == null) return false;

        while(q != null && q.next != null){
            if(p == q) return true;

            p = p.next;
            q = q.next.next;
        }

        return false;

    }
}